src/file/FileDialog.tsx

```diff
@@ -215,6 +218,74 @@ const OpenLocalFile: React.FC<SourceTabProps> = ({ onDismiss }) => {
     );
 };
 
+const ImportFromString: React.FC<SourceTabProps> = ({ onDismiss }) => {
+    const loadScene = useLoadScene();
+    const setSavedState = useSetSavedState();
+    const isDirty = useIsDirty();
+    const theme = useTheme();
+    const [data, setData] = useState<string | undefined>('');
+    const [error, setError] = useState<string | undefined>('');
+
+    const importCallback = useCallback(async () => {
+        if (!data) {
+            return;
+        }
+
+        if (isDirty) {
+            if (!(await confirmUnsavedChanges(theme))) {
+                return;
+            }
+        }
+
+        let scene;
+        try {
+            scene = await textToScene(data);
+        } catch (ex) {
+            console.log(ex);
+            setError('Invalid Import String');
+            return;
+        }
+
+        loadScene(scene, undefined);
+        setSavedState(scene);
+        onDismiss?.();
+    }, [data, isDirty, theme, loadScene, setSavedState, onDismiss]);
+
+    const onKeyChange = useCallback(
```

Comment:
Should this be named `onChange`? Looks like it's handling a change to the text value, not a change in anything related to a key.